apps/electron/layers/main/src/handlers/db/ensure-db.ts

```diff
@@ -1,94 +1,130 @@
-import { watch } from 'chokidar';
+import { watch } from '@affine/native';
+import type { NotifyEvent } from '@affine/native/event';
 import { app } from 'electron';
+import {
+  fromEvent,
+  identity,
+  lastValueFrom,
+  NEVER,
+  Observable,
+  Subject,
+} from 'rxjs';
+import {
+  debounceTime,
+  distinctUntilChanged,
+  exhaustMap,
+  filter,
+  groupBy,
+  ignoreElements,
+  map,
+  mergeMap,
+  shareReplay,
+  startWith,
+  take,
+  takeUntil,
+  tap,
+} from 'rxjs/operators';
 
 import { appContext } from '../../context';
 import { subjects } from '../../events';
 import { logger } from '../../logger';
-import { debounce, ts } from '../../utils';
+import { ts } from '../../utils';
 import type { WorkspaceSQLiteDB } from './sqlite';
 import { openWorkspaceDatabase } from './sqlite';
 
-const dbMapping = new Map<string, Promise<WorkspaceSQLiteDB>>();
-const dbWatchers = new Map<string, () => void>();
+const databaseInput$ = new Subject<string>();
+const terminate$ = app ? fromEvent(app, 'before-quit') : NEVER;
+
+const database$ = databaseInput$.pipe(
```

Comment:
I think we may still need to have mutliple db's to be watched at the same time soon later, e.g. we have multiple-tab/window and each of them need to watch for db file changes.

Reply:
The watch logics are not changed at all; the `database$` emits different `workspaceId` from the upstream

Reply:
See my comment below